File: data/src/main/java/com/bunbeauty/data/State.kt
package com.bunbeauty.data

sealed class State<T> {
    class Loading<T> : State<T>()
    data class Data<T>(val data: T) : State<T>()
    class Empty<T> : State<T>()
    data class Error<T>(val error: String) : State<T>()
}

File: PapaKarloSwift/PapaKarloSwift/state/State.swift
//
//  State.swift
//  PapaKarloSwift
//
//  Created by Valentin on 07.05.2022.
//

import Foundation

enum StateUI<T> {
    
    case loading
    case success(T)
    case empty
    case error(String)
    
}
